fix(config): restore custom logger feature

Qard opened the pull request:
This PR restores the custom logger feature.
Closes elastic/apm-agent-nodejs#289

watson commented on lib/agent.js:
```diff
@@ -5,7 +5,7 @@ var parseUrl = require('url').parse
 
 var afterAll = require('after-all-results')
-var debug = require('debug')('elastic-apm')
+var ConsoleLogLevel = require('console-log-level')
```
It's kind of a convention that variables that start with an uppercase letter like this needs to be instantiated with the `new` keyword, so I'd probably change it to `consoleLogLevel`

Qard replied:
fixed

watson commented on lib/agent.js:
```diff
@@ -68,7 +67,11 @@ Agent.prototype._config = function (opts) {
+  this.logger = this._conf.logger || ConsoleLogLevel
+
+  if (typeof this.logger === 'function') {
+    this.logger = this.logger({ level: this._conf.logLevel })
```
Is it safe to assume that all loggers that are a function will have this API?

Qard replied:
I meant for this to just be a possible initialization option. If you have a logger that doesn't initialize this way, you can just initialize it externally and provide the constructed object instead.

watson replied:
I'm thinking if a simpler API simply would be to require people to initialize their loggers before hand?

watson commented on lib/parsers.js:
```diff
@@ -39,7 +38,8 @@ exports.parseMessage = function (msg) {
   stackman.callsites(err, function (_err, callsites) {
     if (_err) {
-      debug('error while getting error callsites: %s', _err.message)
+      console.log(agent)
```
Ups 😄

Qard replied:
fixed

watson commented on lib/instrumentation/shimmer.js:
```diff
@@ -15,7 +15,7 @@
  */
 
-var debug = require('debug')('elastic-apm')
+var agent = require('../agent').singleton
```
Why not `require('../../')`?

Qard replied:
fixed

watson commented on test/parsers.js:
```diff
@@ -381,6 +381,12 @@ test('#parseError()', function (t) {
         sourceLinesErrorAppFrames: 5,
         sourceLinesErrorLibraryFrames: 5
+      },
+      logger: {
```
Consider DRY'ing this up (is the DRY ruby'ism known in outside of the Ruby world?)

Qard replied:
fixed

watson commented on test/queue.js:
```diff
@@ -7,7 +7,13 @@ var Queue = require('../lib/queue')
     maxQueueSize: 5,
-    flushInterval: 1e6
+    flushInterval: 1e6,
+    logger: {
```
Consider DRY'ing this up

Qard replied:
fixed

Qard commented:
I changed it to not have the logger-as-a-function thing, so the level can just be controlled externally. The docs already partially suggested that in the custom logger section, so I just added an extra note to the `logLevel` option.
